distribution/lib/Standard/Base/0.0.0-dev/src/Internal/Time/Format/Analyzer.enso

```diff
@@ -0,0 +1,137 @@
+        Problem_Behavior.Report_Warning.attach_problems_after continuation problem_builder.to_vector
+
+    ## PRIVATE
+       Prepares a list of warnings that are only reported when parsing using the
+       formatter.
+    get_parsing_only_warnings : Vector
+    get_parsing_only_warnings self =
+        problem_builder = Vector.new_builder
+        self.check_missing_ampm_in_hour_parse problem_builder
+        self.check_missing_year_in_date_parse problem_builder
+        problem_builder.to_vector
+
+    ## PRIVATE
+    check_possible_m_mismatches self problem_builder =
+        pattern_nodes = self.flattened
+        pattern_nodes.each_with_index ix-> value-> case value of
+            Standard_Date_Patterns.Month _ ->
+                # Warns only if surrounded from both sides or if it has a time node on the left and is the last node.
+                has_time_on_left = pattern_nodes.get ix-1 . is_a Time_Patterns
+                has_time_on_right_or_is_last =
+                    next = pattern_nodes.get ix+1
+                    next.is_nothing || next.is_a Time_Patterns
+                if has_time_on_left && has_time_on_right_or_is_last then
+                    problem_builder.append (Suspicious_Date_Time_Format.Warning "A Month pattern 'M' is used next to time patterns. Did you mean 'm' for minutes? (You can remove this warning using `remove_warnings Suspicious_Date_Time_Format`.)")
+
+            Time_Patterns.Minute _ ->
+                has_date_on_both_sides = (pattern_nodes.get ix-1 . is_a Standard_Date_Patterns) && (pattern_nodes.get ix+1 . is_a Standard_Date_Patterns)
+                if has_date_on_both_sides then
+                    problem_builder.append (Suspicious_Date_Time_Format.Warning "A Minute pattern 'm' is used between date patterns. Did you mean 'M' for months? (You can remove this warning using `remove_warnings Suspicious_Date_Time_Format`.)")
+
+            _ -> Nothing
+
+    ## PRIVATE
+    check_possible_seconds_aliasing self problem_builder =
+        pattern_nodes = self.flattened
+        seconds = pattern_nodes.filter node-> case node of
+            Time_Patterns.Second _ -> True
+            _ -> False
+        if seconds.length == 2 then
+            problem_builder.append (Suspicious_Date_Time_Format.Warning "Two Second patterns have been detected ('s'/'S'). Our simple format treats seconds in a case-insensitive way. If you want to indicate a fraction of a second, use 'f' instead. (You can remove this warning using `remove_warnings Suspicious_Date_Time_Format`.)")
+
+    ## PRIVATE
+    has_24h : Boolean
+    has_24h self =
+        self.flattened.any node-> case node of
+            Time_Patterns.Hour _ is24h -> is24h
+            _ -> False
+
+    ## PRIVATE
+    has_12h : Boolean
+    has_12h self =
+        self.flattened.any node-> case node of
+            Time_Patterns.Hour _ is24h -> is24h.not
+            _ -> False
+
+    ## PRIVATE
+    has_ampm : Boolean
+    has_ampm self =
```

Comment:
Should we use `am_pm` throughout?
```suggestion
    has_am_pm : Boolean
    has_am_pm self =
```

Reply:
It's just internal stuff, but yeah I guess that will be better - I will amend it.